modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/DefaultSourceCallbackContext.java

```diff
@@ -108,6 +111,7 @@ public TransactionHandle bindConnection(Object connection) throws ConnectionExce
         transactionHandle = DEFAULT_TRANSACTION_HANDLE;
       }
     } catch (Exception e) {
+      LOGGER.warn("Connection could not be bound", e);
```

Comment:
Comment to reviewer: I added this logger because if we/the customer had it, I think we would have had way more important information and it would be quite straight forward that ClassCastException --> Not rolling back --> not being able to bind new tx. 

This one would have also been helpful for the previous issue I saw with Global Error handling (skill issues from my side, I temporarily added the log but then when cleaning up I removed it)